app/poi_api/poi_suggestion.py

```diff
@@ -1,28 +1,66 @@
 from datetime import datetime
+from app.poi_api.errors import POINotFoundError, InvalidPOISuggestionError
+from app.common import BadDataError
+from typing import Dict
+import json
 
 
-class POI_suggestions:
+class POI_suggestion:
     def __init__(
-        self, pid, suggestion_name, notes, submission_time, submitted_by
+        self,
+        pid,
+        suggestion_name,
+        notes,
+        submitted_by,
+        submission_time=datetime.now(),
     ) -> None:
-        self.pid = pid
+        self._pid = pid
         self.suggestion_name = suggestion_name
         self.notes = notes
-        self.submission_time = submission_time
         self.submitted_by = submitted_by
+        self.submission_time = submission_time
+
+    def get_pid(self) -> str:
+        """Return pid of the POI_suggestion"""
+        return self._pid
+
+    @staticmethod
+    def from_dict(dict: Dict[str, str]) -> "POI_suggestion":
+        """
+        Creates a new POI_suggestion object from a Python Dictionary
+
+        Args:
+            dict: Dictionary of key-value pairs corresponding to user.
+
+        Returns:
+            User: from specified data
+
+        Raises:
+            BadDataError: If required data is missing from the dictionary
+        """
+        try:
+            return POI_suggestion(
+                dict["_pid"],
+                dict["suggestion_name"],
+                dict["notes"],
+                dict["submission_time"],

```

Comment:
This isn't gonna parse correctly from time to datetime. 
`datetime.fromtimestamp(dict["submission_time"].timestamp())` is what you want.

Reply:
I think it would be better to store all of our timestamps as datetimes (Firestore supports storing dates and the Python library supports datetime for this), and format timestamps in ISO 8601 format instead whenever we need to serialize into a string (since this is a standard and widely used format, and converting to/from a datetime in this format is easy)